File: jan6/jan6.cpp
#include <iostream>
#include <unordered_map>

char firstNonRepeatingChar(std::string s) {
    std::unordered_map<char, int> charMap;
    for (char c : s) {
        charMap[c]++;
    }
    for (char c : s) {
        if (charMap[c] == 1) {
            return c;
        }
    }
    return ' ';
}

int main() {
    std::string input = "leetcode";
    char result = firstNonRepeatingChar(input);
    std::cout << result << std::endl;
    return 0;
}

File: jan6/jan6.cs
using System;
using System.Collections.Generic;

class Program {
    public static char FirstNonRepeatingChar(string s) {
        Dictionary<char, int> charMap = new Dictionary<char, int>();
        foreach (char c in s) {
            if (charMap.ContainsKey(c)) {
                charMap[c]++;
            } else {
                charMap[c] = 1;
            }
        }
        foreach (char c in s) {
            if (charMap[c] == 1) {
                return c;
            }
        }
        return ' ';
    }

    static void Main() {
        string input = "leetcode";
        char result = FirstNonRepeatingChar(input);
        Console.WriteLine(result);
    }
}